packages/mobile/src/screens/stake/undelegate/index.tsx

```diff
@@ -156,12 +161,68 @@ export const UndelegateScreen: FunctionComponent = observer(() => {
       */}
       <AmountInput label="Amount" amountConfig={sendConfigs.amountConfig} />
       <MemoInput label="Memo (Optional)" memoConfig={sendConfigs.memoConfig} />
-      <FeeButtons
-        label="Fee"
-        gasLabel="gas"
-        feeConfig={sendConfigs.feeConfig}
-        gasConfig={sendConfigs.gasConfig}
-      />
+      <View
+        style={{
+          flexDirection: 'row',
+          paddingBottom: 24,
+          alignItems: 'center'
+        }}
+      >
+        <Toggle
+          on={customFee}
+          onChange={value => {
+            setCustomFee(value);
+            if (!value) {
+              if (
+                sendConfigs.feeConfig.feeCurrency &&
+                !sendConfigs.feeConfig.fee
+              ) {
+                sendConfigs.feeConfig.setFeeType('average');
+              }
+            }
+          }}
+        />
+        <Text
+          style={{
+            fontWeight: '700',
+            fontSize: 16,
+            lineHeight: 34,
+            paddingHorizontal: 8
+          }}
+        >
+          Custom Fee
+        </Text>
+      </View>
+      {customFee && chainStore.current.networkType !== 'evm' ? (
+        <TextInput
+          label="Fee"
+          placeholder="Type your Fee here"
+          keyboardType={'numeric'}
+          labelStyle={{
+            fontSize: 16,
+            fontWeight: '700',
+            lineHeight: 22,
+            color: colors['gray-900'],
+            marginBottom: spacing['8']
+          }}
+          onChangeText={text => {
+            const fee = new Dec(Number(text.replace(/,/g, '.')) * 10 ** 6);
```

Comment:
`new Dec(Number(text.replace(/,/g, '.')).mul(Dec.getPrecisionDec(6))`

Reply:
Number does not have mul(Dec.getPrecisionDec(6), so i replace it with Math.pow